lib/services/livesync/livesync-service.ts

```diff
@@ -120,6 +135,150 @@ export class LiveSyncService extends EventEmitter implements ILiveSyncService {
 		this.$logger.info(`Successfully synced application ${liveSyncResultInfo.deviceAppData.appIdentifier} on device ${liveSyncResultInfo.deviceAppData.device.deviceInfo.identifier}.`);
 	}
 
+	private async refreshApplicationWithDebug(projectData: IProjectData, liveSyncResultInfo: ILiveSyncResultInfo, debugOptions: IDebugOptions, outputPath?: string): Promise<void> {
+		await this.$platformService.trackProjectType(projectData);
+
+		const deviceAppData = liveSyncResultInfo.deviceAppData;
+
+		const deviceIdentifier = liveSyncResultInfo.deviceAppData.device.deviceInfo.identifier;
+		await this.$debugService.debugStop(deviceIdentifier);
+
+		let applicationId = deviceAppData.appIdentifier;
+		const attachDebuggerOptions: IAttachDebuggerOptions = {
+			platform: liveSyncResultInfo.deviceAppData.device.deviceInfo.platform,
+			isEmulator: liveSyncResultInfo.deviceAppData.device.isEmulator,
+			projectDir: projectData.projectDir,
+			deviceIdentifier,
+			debugOptions,
+			outputPath
+		};
+
+		try {
+			await deviceAppData.device.applicationManager.stopApplication(applicationId, projectData.projectName);
+		} catch (err) {
+			this.$logger.trace("Could not stop application during debug livesync. Will try to restart app instead.", err);
+			if ((err.message || err) === "Could not find developer disk image") {
+				await this.refreshApplicationWithoutDebug(projectData, liveSyncResultInfo, debugOptions, outputPath, { shouldSkipEmitLiveSyncNotification: true });
+				this.emit(USER_INTERACTION_NEEDED_EVENT_NAME, attachDebuggerOptions);
+				return;
+			} else {
+				throw err;
+			}
+		}
+
+		const deviceOption = {
+			deviceIdentifier: liveSyncResultInfo.deviceAppData.device.deviceInfo.identifier,
+			debugOptions: debugOptions,
+		};
+
+		return this.enableDebuggingCoreWithoutWaitingCurrentAction(deviceOption, { projectDir: projectData.projectDir });
+	}
+
+	public async attachDebugger(settings: IAttachDebuggerOptions): Promise<void> {
+		settings.debugOptions = settings.debugOptions || {
+			chrome: true,
+			start: true
+		};
+		const projectData = this.$projectDataService.getProjectData(settings.projectDir);
+		let debugData = this.$debugDataService.createDebugData(projectData, { device: settings.deviceIdentifier });
+
+		// Of the properties below only `buildForDevice` and `release` are currently used.
+		// Leaving the others with placeholder values so that they may not be forgotten in future implementations.
+		const buildConfig: IBuildConfig = {
+			buildForDevice: !settings.isEmulator,
+			release: false,
+			device: settings.deviceIdentifier,
+			provision: null,
```

Comment:
This `buildConfig` isn't actually used for building - only for getting the last output path